### PR TITLE
Move MediaTrack ToProto out of media track receiver

### PR DESCRIPTION
**Pull request by boks1971**
as remote needs a different implementation.

In the never ending chase, have to move this to play well with remote :-)

### CODE REVIEW

File: pkg/rtc/localparticipant.go [hunk 1]
```diff
@@ -229,7 +229,7 @@ func (l *LocalParticipant) MediaTrackReceived(track *webrtc.TrackRemote, rtpRece
 			SubscriberConfig:    l.params.Config.Subscriber,
 		})
 
-		for ssrc,info  :=range l.params.SimTracks {
+		for ssrc, info := range l.params.SimTracks {
```
Comment (boks1971) [inline pkg/rtc/localparticipant.go:232]: just go fmt stuff

File: pkg/rtc/mediatrack.go [hunk 2]
```diff
@@ -221,6 +238,12 @@ func (t *MediaTrack) AddReceiver(receiver *webrtc.RTPReceiver, track *webrtc.Tra
 	})
 }
 
+func (t *MediaTrack) TrySetSimulcastSSRC(layer uint8, ssrc uint32) {
```
Comment (boks1971) [inline pkg/rtc/mediatrack.go:241]: @cnderrauber Saw some discussion between you and @davidzhao on lock for track info. I saw that `layerSsrcs` did not seem to have a lock before also and so does `ToProto`. So, just keeping them without locks and moving it into `MediaTrack` as `RemoteMediaTrack` does not have this.

Comment (cnderrauber) [reply pkg/rtc/mediatrack.go:241]: Thanks!

**Review by cnderrauber — APPROVED** [review]
looks good!

**Pull request merged**